tests/dbus/_misc.py

```diff
@@ -20,25 +20,94 @@
 import os
 import string
 import subprocess
+import time
 
 from hypothesis import strategies
 
 _STRATISD = os.environ['STRATISD']
 
+SIMULATOR = 'sim'
+MODE = os.getenv('STRATIS_TEST_MODE', SIMULATOR).lower()
+DEVICES = os.getenv('STRATIS_TEST_DEVICES', "").split()
+
+
+ZERO_BUFFER = bytearray(1024*8)
+
+
+def wipe_stratus_device(device):
+    """
+    Wipe the first 8K of the specified device
+    :param device: Block device to clear
+    :return: None
+    """
+    # TODO Check to see if this is a stratis device first?
+    with open(device, "wb") as disk:
+        disk.write(ZERO_BUFFER)
+
+
+def get_processes(cmd_search_string):
+    """
+    Walk /proc looking for processes that contain cmd_search_string in the
+    cmdline.
+    :param cmd_search_string:
+    :return: List of tuples (pid, cmdline)
+    """
+    rc = []
+    pids = [pid for pid in os.listdir('/proc') if pid.isdigit()]
+    for p in pids:
+        try:
+            cmdline = open(os.path.join('/proc', p, 'cmdline')).read()
+            if cmd_search_string in cmdline:
+                rc.append((int(p), cmdline))
+
+        except IOError:
+            pass
+    return rc
+
+
+class RealDevice(object):
+    """
+    Offer the same API with real devices as is present for hypothesis
+    """
+    def __init__(self, devices=None):
+        self.device_list = devices
+
+    def example(self):
+        return self.device_list
+
 
 def _device_list(minimum):
     """
     Get a device generating strategy.
 
     :param int minimum: the minimum number of devices, must be at least 0
     """
-    return strategies.lists(
-       strategies.text(
-          alphabet=string.ascii_letters + "/",
-          min_size=1
-       ),
-       min_size=minimum
-    )
+    if MODE == SIMULATOR:
+        return strategies.lists(
+           strategies.text(
+              alphabet=string.ascii_letters + "/",
+              min_size=1
+           ),
+           min_size=minimum
+        )
+
+    return RealDevice(DEVICES)
+
+
+def ensure_no_daemon():
```

Comment:
This method and call doesn't have anything to do with using actual devices, AFAICT. It seems like it deserves a separate commit.